Backpack-SwiftUI/FlightLeg/Classes/BPKFlightLeg.swift

```diff
@@ -0,0 +1,177 @@
+import SwiftUI
+import Backpack_Common
+
+struct BPKFlightLeg: View {
+    private let airlineLogoSize: CGFloat = 24.0
+
+    private let departureArrivalTime: String
+    private let nextDayArrival: String?
+    private let flightDescription: AttributedString
+    private let stopsInfo: String
+    private let highlightStopsInfo: Bool
+    private let duration: String
+    private let operatedBy: String?
+    private let warning: String?
+    @State private var carrierLogo: Image?
+
+    // String-based init
+    public init(
+        departureArrivalTime: String,
+        nextDayArrival: String? = "",
+        flightDescription: String,
+        stopsInfo: String,
+        highlightStopsInfo: Bool,
+        duration: String,
+        operatedBy: String? = "",
+        warning: String? = "",
+        carrierLogo: UIImage? = nil
+    ) {
+        self.departureArrivalTime = departureArrivalTime
+        self.nextDayArrival = nextDayArrival
+        self.flightDescription = AttributedString(flightDescription)
+        self.stopsInfo = stopsInfo
+        self.highlightStopsInfo = highlightStopsInfo
+        self.duration = duration
+        self.operatedBy = operatedBy
+        self.warning = warning
+        if let carrierLogo = carrierLogo {
+            self._carrierLogo = State(initialValue: Image(uiImage: carrierLogo))
+        } else {
+            self._carrierLogo = State(initialValue: nil)
+        }
+    }
+
+    // AttributedString-based init
+    public init(
+        departureArrivalTime: String,
+        nextDayArrival: String? = "",
+        flightDescription: AttributedString,
+        stopsInfo: String,
+        highlightStopsInfo: Bool,
+        duration: String,
+        operatedBy: String? = "",
+        warning: String? = "",
+        carrierLogo: UIImage? = nil
+    ) {
+        self.departureArrivalTime = departureArrivalTime
+        self.nextDayArrival = nextDayArrival
+        self.flightDescription = flightDescription
+        self.stopsInfo = stopsInfo
+        self.highlightStopsInfo = highlightStopsInfo
+        self.duration = duration
+        self.operatedBy = operatedBy
+        self.warning = warning
+        if let carrierLogo = carrierLogo {
+            self._carrierLogo = State(initialValue: Image(uiImage: carrierLogo))
+        } else {
+            self._carrierLogo = State(initialValue: nil)
+        }
+    }
+
+    var body: some View {
+        BPKDynamicStack(horizontalAlignment: .leading, verticalAlignment: .top, spacing: .base) {
+            createCarrierLogo()
+            createFlightInfo()
+                .frame(
+                    maxWidth: .infinity,
+                    alignment: .leading
+                )
+            createDurationInfo()
+        }
+    }
+
+    @ViewBuilder
+    private func createCarrierLogo() -> some View {
+        VStack(alignment: .leading, spacing: .sm) {
+            if let carrierLogo = carrierLogo {
```

Comment:
this if let should be a parent of the VStack I believe, else it will draw an empty VStack, which doesnt seem to make a loot of sense (maybe it does because of the padding.top you're setting?)

Reply:
Also, this if let can be written like this: 
```swift
if let carrierLogo { // if the property name is the same it can be ignored!
}